### PR TITLE
Add support show tables and show columns for ballista

### PR DESCRIPTION
**Pull request by gaojun2048**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/1592.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Looks good to me -- thank you @gaojun2048

File: ballista/rust/client/src/context.rs [hunk 1]
```diff
@@ -256,6 +288,14 @@ impl BallistaContext {
             )
         };
 
+        let is_show = self.is_show_statement(sql).await?;
```
Comment (alamb) [inline ballista/rust/client/src/context.rs:291]: I wonder if there is some reason not to always use the `with_information_schema`?

**Review by alamb — APPROVED** [review]

File: ballista/rust/client/src/context.rs (hunk 1 above)
Comment (Ted-Jiang) [reply ballista/rust/client/src/context.rs:291]: Is it ok set ` information_schema: true,` by default ? if not may i ask which situation need block this for user  😊
https://github.com/apache/arrow-datafusion/blob/82e80036d9dbcf1bacac8e94eb10d5c60a8016a7/datafusion/src/execution/context.rs#L930

**Review by Ted-Jiang — APPROVED** [review]

File: ballista/rust/client/src/context.rs (hunk 1 above)
Comment (gaojun2048) [reply ballista/rust/client/src/context.rs:291]: Datafusion supports show tables only when with_information_schema=true. Currently, users cannot specify the with_information_schema parameter of datafusion when creating a ballista context. I can add a with_information_schema parameter in BallistaConfig, the default value is false, when users need to use show tables in ballista, users can turn on this function through with_information_schema(true)
whether

**Pull request merged**